tests/trainer/test_checkpoint.py

```diff
@@ -461,8 +507,70 @@ def test_load_weights(self, device, load_weights_only):
         if load_weights_only:
             # callback state should not have been loaded
             assert not stateful_callbacks_equal
+            assert not metrics_equal
         else:
             assert stateful_callbacks_equal
+            assert metrics_equal
+
+    @pytest.mark.remote
+    @device('cpu')
+    @pytest.mark.parametrize('load_weights_only', [True, False])
+    def test_load_remote_checkpoint(self, device, tmp_path: pathlib.Path, load_weights_only):
+        """
+        This test checks if our checkpointing is backwards compatible.
+        We should be able to load in a saved checkpoint and continue training.
+        The checkpoint weight and metrics should match at load time
+        and should be equivalent after training continues.
+        Checkpoint saved using: Composer 0.13.5 with default dependencies.
+        """
+        import os
+
+        trainer_1 = self.get_trainer(save_folder='first', device=device)
+        trainer_1.fit()
+        trainer_1.close()
+
+        remote_checkpoint = 's3://mosaicml-internal-checkpoints-test/backwards_compatibility/trained_cpu_checkpoint_ep2.pt'
```

Comment:
lets parametrize this path? then we can easily add new checkpoints to it over time? better yet, we would ls a bucket and do all of them, but lets defer that for now :) also sorry, why doesn't this work with oci? it should.

Reply:
oh jk, cause our pytest suite doesn't have an oci integration, sg